GltfExport.cs

```diff
@@ -64,93 +68,103 @@ protected override void SolveInstance(IGH_DataAccess DA)
-                        {
-                            LayerIndex = 0,
-                            MaterialSource = Rhino.DocObjects.ObjectMaterialSource.MaterialFromLayer
-                        };
-                        foreach (var branch in geo!.Branches)
-                            foreach (var g in branch)
+                            // material assignment and baking
+                            int path = 0;
+                            foreach (var branch in geo!.Branches)
                             {
-                                g.CastTo<GeometryBase>(out GeometryBase geometry);
-                                ids.Add(doc.Objects.Add(geometry, att));
-                            }
-                        RhinoApp.Write("Object(s) baked\n", true);
-
-                        // select the object(s)
-                        foreach (var id in ids)
-                            doc.Objects.Select(id, true);
-                        RhinoApp.Write("Object(s) selected\n", true);
-
-                        // Assign the material(s) to the object(s) if exist(s)
-                        int b = 0;
-                        int i = 0;
-                        foreach (var branch in material.Branches)
-                        {
-                            foreach (var m in branch)
-                            {
-                                var materialRH = doc.RenderMaterials.FirstOrDefault(n => n.Name == m.Value);
-                                if (materialRH != null)
+                                var matLst = material.Branches[RepeatLast(material.Branches.Count, path)]; // get mat branch list (repeating last if less)
+                                int index = 0;
+                                RenderMaterial? materialRH = null;
+                                foreach (var g in branch)
                                 {
-                                    var rhObj = doc.Objects.FindId(ids[(b * i++) % geo.Paths[b].Length]); //TODO: check if that math works
-                                    rhObj.RenderMaterial = materialRH;
-                                    rhObj.CommitChanges();
-                                    doc.Views.Redraw();
-                                    RhinoApp.Write("Object material assigned\n", true);
+                                    // assigning custom materials if exists (repeating last item if less)
+                                    if (matLst != null)
+                                    {
+                                        var matName = matLst[RepeatLast(matLst.Count, index)].Value;
+                                        materialRH = doc.RenderMaterials
+                                            .FirstOrDefault(n => n.Name == matName); // get render mat
+                                    }
+
+                                    // baking
+                                    Guid id = doc.Objects.Add(GH_Convert.ToGeometryBase(g), new ObjectAttributes()
```

Comment:
# 👍 Material Assignment and Baking
 - assigning materials dynamically based on the presence of objects
 - using RenderMaterial property assignment for object attributes
  - clever!